sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -256,3 +267,27 @@ def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] =
         token_info = cast(SupportsTokenInfo, super()).get_token_info(*scopes, options=options)
         within_dac.set(False)
         return token_info
+
+
+def parse_azure_dac(az_dac):
+    striped_az_dac = az_dac.strip()
+    if striped_az_dac.endswith(";"):
+        striped_az_dac = striped_az_dac[:-1]
+    creds = [cred.strip() for cred in striped_az_dac.split(";")]
+    return creds
+
+
+def resolve_credentials(creds, avail_credentials):
+    credentials = []
+
+    for cred in creds:
+        if cred not in avail_credentials:
+            raise ValueError(
```

Comment:
I think we have scenarios where raising here could be problematic. For example, if a user runs the following code on AKS, it would work. However, running this locally, the user would encounter this `ValueError` since `WorkloadIdentityCredential` was not added to avail_credentials due to the lack of certain environment variables.

```python
cred = DefaultAzureCredential(default_credential_allow_list=["ENVIRONMENT", "WORKLOAD_IDENTITY", "CLI"])
```

I think we either need to update WorkloadIdentityCredential to not raise in the constructor (instead raise in `get_token`) and then always add it to avail_credentials. Or not raise here .

Reply:
Good catch!